modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/data/provider/AnalyticsReportsDataProvider.java

```diff
@@ -153,6 +154,76 @@ public HistoricalMetric getHistoricalViewsHistoricalMetric(
 		}
 	}
 
+	public List<ReferringURL> getReferringDomains(long companyId, String url)
+		throws PortalException {
+
+		try {
+			String response = _asahFaroBackendClient.doGet(
+				companyId,
+				"api/1.0/pages/page-referrer-hosts?canonicalURL=" + url +
+					"&interval=D&rangeKey=30");
+
+			TypeFactory typeFactory = _objectMapper.getTypeFactory();
+
+			Map<String, Long> socialPageReferrers = _objectMapper.readValue(
```

Comment:
`socialPageReferrers` sounds a bit inaccurate for me. Maybe something like `pageReferrerHosts`? What do you think?